File: src/main/java/service/UsuarioService.java
package service;

import dao.UsuarioDao;
import kontroller.Usuario;
import service.exception.ServiceException;

import javax.inject.Inject;
import java.io.Serializable;

public class UsuarioService extends ServiceAbstract<Usuario> implements Serializable {

    private static final long serialVersionUID = -2800411759544395063L;
    private UsuarioDao dao;

    @Inject
    public UsuarioService(UsuarioDao dao) {
        super(dao);
        this.dao = dao;
    }

    public Usuario findUserWith(String username, String password) {
        return dao.findUserWith(username, password);
    }
}

File: src/main/java/service/UsuarioService.kt
package service

import dao.UsuarioDao
import model.Usuario
import java.io.Serializable
import javax.inject.Inject

class UsuarioService @Inject constructor(var daoU: UsuarioDao) : Serializable, AbstractService<Usuario>(dao = daoU) {

    val serialVersionUID = -2800411759544395063L

    fun findUserWith(username: String, password: String): Usuario? {
        return daoU.findUserWith(username, password)
    }
}


